qiskit_ibm_runtime/session.py

```diff
@@ -78,6 +78,7 @@ def __init__(
         service: Optional[QiskitRuntimeService] = None,
         backend: Optional[Union[str, IBMBackend]] = None,
         max_time: Optional[Union[int, str]] = None,
+        mode: Optional[str] = None,
```

Comment:
`mode` shouldn't be a externally visible parameter. I.e. we should not allow `Session(mode="batch")`.